src/analysis/inner_product_space/positive.lean

```diff
@@ -40,15 +40,16 @@ open_locale inner_product complex_conjugate
 namespace continuous_linear_map
 
 variables {𝕜 E F : Type*} [is_R_or_C 𝕜] [inner_product_space 𝕜 E] [inner_product_space 𝕜 F]
+  [complete_space E] [complete_space F]
 local notation `⟪`x`, `y`⟫` := @inner 𝕜 _ _ x y
 
 /-- A continuous linear endomorphism `T` of a Hilbert space is **positive** if it is self adjoint
   and `∀ x, 0 ≤ re ⟪T x, x⟫`. -/
 def is_positive (T : E →L[𝕜] E) : Prop :=
-  is_self_adjoint (T : E →ₗ[𝕜] E) ∧ ∀ x, 0 ≤ T.re_apply_inner_self x
+  is_self_adjoint T ∧ ∀ x, 0 ≤ T.re_apply_inner_self x
```

Comment:
Is this a regression?  This seems to me like a situation where it would be convenient to keep using `is_symmetric` rather than switch to the new `is_self_adjoint`.  The current definition requires `E →L[𝕜] E` but the early lemmas will work for `E →ₗ[𝕜] E` and it seems like we should eventually switch to that setting.  Unnecessary dependence on the adjoint here will make that impossible.

(If I missed an earlier discussion of this point, feel free to point me to it.)

Reply:
Of course the lemmas work for both cases, they are equivalent. My point was if the operator is a-priori defined as a continuous linear operator, then one should use `is_self_adjoint` and if it is defined as a linear operator, then `is_symmetric` is the correct definition (as there is in general (i.e., infinite dimension) no adjoint).

Reply:
I think I agree with Heather here. This is the *concrete* definition of positivity, so the included selfadjoint condition should really be `is_symmetric` defined in terms of the inner product. Regressions aside, at the very least, if you're proving an operator is positive this way, it will likely be easy to show that it is symmetric (because you apparently have control over the values of the quadratic form), and then you don't need to invoke Hellinger-Toeplitz to get the selfadjoint condition.

Reply:
Note that in the definition we assume that the operator is continuous. We could add a definition that constructs positive operator from `linear_map` that is symmetric and satisfies `∀ x, 0 ≤ T.re_apply_inner_self x`. That would be the easiest way to construct a positive operator. I guess there is literature that proves the boundedness and the symmetry, but this is obviously redundant.

Reply:
I think maybe we are talking past each other, but I can't quite tell. Why not just have `linear_map.is_positive` defined as `is_symmetric` and `∀ x, 0 ≤ T.re_apply_inner_self x`? Then it would apply in both situations (admittedly, dot notation wouldn't work so well anymore in the continuous case).

However, if your point is that `is_positive` should really be defined as `is_self_adjoint` and `∀ x, 0 ≤ T.re_apply_inner_self x` for `linear_pmaps` in the unbounded case (as opposed to using `is_symmetric`), then I would prefer any definition which would unify the three settings, although I have some doubts whether this is really possible.

Reply:
I agree that we might be not talking about the same thing. The current `is_positive` is defined for `continuous_linear_map`, so my point is that there is no reason to prefer `is_symmetric` over `is_self_adjoint`.
If it was defined for `linear_map`, then I would agree that `is_symmetric` is the correct choice. I don't think that changing the definition of `is_positive` from `continuous_linear_map` to `linear_map` is in the scope of this PR (I think the definition as it is now is good). Also, I don't really see the value in adding a definition `linear_map.is_positive`, but maybe you have a good reason for that. There might be an argument that it is necessary for the finite dimensional case, because we don't want to talk about `continuous_linear_map`s in finite dimensions.
As for the `linear_pmap` I cannot guarantee anything, but I am rather optimistic that we will need only two definitions (we could get away with one, but this would cause lots of annoying things with coercing to `linear_pmap`).

I am sorry if I come across as too stubborn on this, I only think that reverting to `is_symmetric` for `continuous_linear_map` would make this whole PR completely useless.